### PR TITLE
Adding User management sdks

### PR DESCRIPTION
**Pull request by yugmamoradia**
Adding the User Management SDKs. This is my first time updating the dotnet sdks, so please let me know if I have missed something or something that should've been in a specific format.

### CODE REVIEW

File: SamplifyAPIClient/Roles.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+using System.Runtime.Serialization;
+
+namespace Dynata.SamplifyAPIClient
+{
+    [DataContract]
+    public class Roles
+    {
+        [DataMember(Name = "id")]
+        public string ID { get; set; }
```
Comment (andyshipman) [inline SamplifyAPIClient/Roles.cs:9]: public string ID --- Should this be a string or an int?

File: SamplifyAPIClient/Teams.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        [DataMember(Name = "id")]
+        public int ID { get; set; }
+        [DataMember(Name = "name")]
+        public string Name { get; set; }
+        [DataMember(Name = "default")]
+        public bool Default { get; set; }
+        [DataMember(Name = "description")]
+        public string Description { get; set; }
+        [DataMember(Name = "status")]
+        public string Status { get; set; }
```
Comment (andyshipman) [inline SamplifyAPIClient/Teams.cs:18]: Can this be defined with constants?

File: SamplifyAPIClient/Teams.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using System.Runtime.Serialization;
+
+namespace Dynata.SamplifyAPIClient
+{
+    [DataContract]
+    public class Team
```
Comment (andyshipman) [inline SamplifyAPIClient/Teams.cs:7]: We have two different class definitions for "Team" (the other is here: SamplifyAPIClient/User.cs). Could one of them be named something different?

**Review by caseylucas — COMMENTED** [review]
@yugmamoradia my experience with this code is practically zero. :( So, I rely on others with more experience for a proper review. (Thanks @andyshipman.) We should be sure to validate that we can use these changes with a .net client program. Andy mentioned that he knew how to run the tests.

File: SamplifyAPIClient/Roles.cs (hunk 1 above)
Comment (yugmamoradia) [reply SamplifyAPIClient/Roles.cs:9]: This should be a string. The example of ID here is `PROJECT_MANAGER`

File: SamplifyAPIClient/Teams.cs (hunk 2 above)
Comment (yugmamoradia) [reply SamplifyAPIClient/Teams.cs:18]: This status here is a part of response. Could you help me understand how can I define a constant here? 
The value of Status here can be either `ACTIVE` or `INACTIVE`.

**Review by andyshipman — APPROVED** [review]

**Review by andyshipman — APPROVED** [review]

**Pull request merged**